ksy_schema.json

```diff
@@ -99,16 +99,23 @@
       "type": "object",
       "additionalProperties": false,
       "properties": {
-        "id": { "$ref": "#/definitions/Identifier" },
-        "title": { "type": "string" },
+        "id": {
+          "$ref": "#/definitions/Identifier",
```

Comment:
I know our schema already does this in other places, but an object with a `$ref` key isn't supposed to contain any other attributes. According to [the JSON Schema Draft 7 spec](https://tools.ietf.org/html/draft-handrews-json-schema-01#section-8.3) (which is what our schema uses currently):

> An object schema with a "$ref" property MUST be interpreted as a "$ref" reference. [...] All other properties in a "$ref" object MUST be ignored.

Reply:
Oops! Totally forgot about that. After quick googling I found [this SO question](https://stackoverflow.com/questions/33565090/json-schema-property-description-and-ref-usage) with [reference to an example](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.7.7.1.1) in the current draft, that is the next (and currently the last) draft. So maybe it's time to switch to it?

Reply:
Yeah, it would probably be a good idea to update our schema to the latest JSON Schema draft. I think it would be better to do that in a separate PR though - we would need to check what exactly has changed that affects us and would require changes, and also check if the current draft is supported by the tools we're using (Docson for the documentation generation in particular).

I think for now it might be okay to leave the other attributes in the `$ref` object here - we already have other places in the spec that do this (even though it violates the draft we currently use), and it will become valid once we update to the latest draft.

Reply:
> Yeah, it would probably be a good idea to update our schema to the latest JSON Schema draft.

I'm not sure, we would need to check if the 2019-09-specific modifications we make in the JSON work in the external tools that often make use of the KSY schema, i.e.:

- [YAML Language Support by Red Hat](https://github.com/redhat-developer/vscode-yaml) (a VS Code plugin; claims to support JSON Schema 7 and below),
- https://github.com/lbovet/docson (used to generate https://doc.kaitai.io/ksy_diagram.html; supposedly supports _JSON schema v4 keywords_, but it has no problem with our `ksy_schema.json` using JSON Schema 7).

But I would be cautious if they do not state explicitly the support for the 2019-09 JSON Schema version. The KSY schema would become pretty useless if it breaks in these tools - like it _can be_ read manually by a human in the raw JSON format, but you know - that sucks.